pkg/models/devops/devops.go

```diff
@@ -143,7 +146,30 @@ func convertToHttpParameters(req *http.Request) *devops.HttpParameters {
 }
 
 func (d devopsOperator) CreateDevOpsProject(workspace string, project *v1alpha3.DevOpsProject) (*v1alpha3.DevOpsProject, error) {
-	project.Annotations[tenantv1alpha1.WorkspaceLabel] = workspace
+	// using generateName instead of devops project name
+	if project.GenerateName == "" {
+		err := errors.NewInvalid(devopsv1alpha3.SchemeGroupVersion.WithKind(devopsv1alpha3.ResourceKindDevOpsProject).GroupKind(),
+			"", []*field.Error{field.Required(field.NewPath("metadata.generateName"), "generateName is required")})
+		klog.Error(err)
+		return nil, err
+	}
+
+	// ensure generateName is unique in workspace scope
+	if unique, err := d.generateNameIsUnique(workspace, project.GenerateName); err != nil {
+		return nil, err
+	} else if !unique {
+		err = errors.NewConflict(devopsv1alpha3.Resource(devopsv1alpha3.ResourceSingularDevOpsProject),
+			project.GenerateName, fmt.Errorf("already exist in workspace scope"))
```

Comment:
The error message should be more informative. 
```suggestion
			project.GenerateName, fmt.Errorf("a devops project named %s already exists in the workspace", project.GenerateName))
```